pennylane/estimator/templates/select.py

```diff
@@ -443,3 +444,289 @@ def controlled_resource_decomp(
             gate_list.append(GateCount(resource_rep(qre.X), 2 * num_zero_ctrl))
 
         return gate_list
+
+
+class SelectPauli(ResourceOperator):
+    r"""Resource class for the ``Select`` opreation used with a Hamiltonian expressed as a linear
+    combination of unitaries (LCU) where each unitary is a Pauli word.
+
+    Args:
+        pauli_ham (:class:`~pennylane.estimator.compact_hamiltonian.PauliHamiltonian`): A Hamiltonian
+            expressed as a linear combination of Pauli words, over which ``Select`` is applied.
+        wires (WiresLike | None): the wires the operation acts on
+
+    Resources:
+        The resources are based on the analysis in `Babbush et al. (2018) <https://arxiv.org/pdf/1805.03662>`_ section III.A,
+        'Unary Iteration and Indexed Operations', and Figures 4, 6, and 7.
+
+        Note: This implementation assumes we have access to :math:`n - 1` additional work qubits,
```

Comment:
"work" qubits  -- is this the right term?

Reply:
Nice catch, changed to auxiliary qubits